projects/openapi-utilities/src/openapi3/implementations/openapi3/log-json-pointer.ts

```diff
@@ -0,0 +1,72 @@
+import { JsonSchemaSourcemap } from '@useoptic/openapi-io';
+import { sourcemapReader } from './sourcemap-reader';
+import chalk from 'chalk';
+
+export function jsonPointerLogger(sourceMap: JsonSchemaSourcemap) {
+  const reader = sourcemapReader(sourceMap);
+
+  return {
+    log: (pointer: string): string | undefined => {
+      const lookupPosition = reader.findFileAndLines(pointer);
+      if (lookupPosition) {
+        const [start, end] = rangeFromLine(lookupPosition.startLine);
+        const source = sourceMap.files.find(
+          (file) => file.path === lookupPosition.filePath
+        );
+        if (!source)
+          throw new Error(
+            'could not render issue for file ' + lookupPosition.filePath
+          );
+
+        const previewLines = source.contents.split('\n').slice(start, end);
+        return renderCodeFrame(
+          start,
+          end,
+          previewLines,
+          lookupPosition.filePath,
+          lookupPosition.startLine
+        );
+      }
+    },
+  };
+}
+
+function rangeFromLine(
+  line: number,
+  before: number = 4,
+  after: number = 3
```

Comment:
I don't understand this - why are these numbers 4 and 3?

Reply:
It's just a preference. Lines above the relevant line and line after. 

it looked nice. I can make them constants

Reply:
Oh I see what it's used for. The function name was a little hard to understand. How about `getDisplayedLineNumbers`